addons/auth_oauth/models/res_users.py

```diff
@@ -37,11 +50,24 @@ def _auth_oauth_validate(self, provider, access_token):
         if oauth_provider.data_endpoint:
             data = self._auth_oauth_rpc(oauth_provider.data_endpoint, access_token)
             validation.update(data)
+        # unify subject key under standard (sub), pop all possible and get most sensible
+        subject = next(filter(None, [
+            validation.pop(key, None)
```

Comment:
It's cleaner to drop the alt subject keys, but don't we create a risk of regression in stable if callers expect their usual key flavor to be present? Perhaps keep them in stable and drop that in master?

```suggestion
            validation.get(key)
```

Reply:
I re-added the value on the `user_id` key as well, as that's the old normalised target I figure it should work well enough.